tests/test_generators/test_pydanticgen.py

```diff
@@ -578,6 +578,59 @@ def test_pydantic_arrays():
     assert temperature_dataset.temperatures_in_K == temperatures
 
 
+def test_pydantic_array_cardinality():
```

Comment:
Maybe just `test_pydantic_cardinality` because `array` is something else in linkml (though I know what you mean) :)